stdlib/public/SDK/CoreGraphics/CGFloat.swift.gyb

```diff
@@ -807,3 +807,17 @@ extension CGFloat : _CVarArgPassedAsDouble, _CVarArgAligned {
     return native._cVarArgAlignment
   }
 }
+
+extension CGFloat : Codable {
+  @_transparent
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.singleValueContainer()
+    self.native = try container.decode(NativeType.self)
```

Comment:
Another thing we can do is attempt to decode `NativeType.self`, and if that fails with `DecodingError.typeMismatch`, attempt the other type. If that fails as well, then it's neither a `Float` nor a `Double`.

I think this is a reasonable middle ground.